### ISSUE TITLE
better magneto switch model

### ISSUE THREAD

**Issue by gilbertohasnofb** [HHS81/c182#21]
We could improve the current magneto switch model, what about using the c172p one? It also has a key attached to it which is animated. What do you think, @HHS81?

**Comment by HHS81** [comment]
at least it needs the key and the animation. I'll take a look in how I can implement this

**Comment by gilbertohasnofb** [comment]
Ok!

**Comment by gilbertohasnofb** [comment]
@HHS81 I think we really need to work on this, right now we can change the magnetos position (OFF L R BOTH START) only using the keyboard (`{` and `}`). Did you think about using the c172p's model, or at least the key?

**Comment by HHS81** [comment]
Together with the correct engine startup procedure,  yes.

**Comment by hbeni** [comment]
Hi i ported the magneto switch from the cp172. It shows its key, starter and selection works as expected.

**Comment by hbeni** [comment]
(partly fixed with PR https://github.com/HHS81/c182s/pull/156)
still open from there is:
- [ ] correct lightmap/uv mapping
- [ ] correct key model (see https://github.com/HHS81/c182s/pull/156#issuecomment-325207929)

**Comment by hbeni** [comment]
The 172 has the correct key since some time.
Someone with 3D skills surely can port this over easily

**Comment by wlbragg** [comment]
@hbeni I'm working on this. Everything is done except figuring out how to add the original alpha channel to my new RGB color lightmap. Hard to explain. I'm hung up on the for channel lightmap. R, G and B channels are good, I edited them and added the new key, but I have to convert or copy the original alpha channel of LMPanel.png to my new R, G and B channels that add the key.
Hopefully I can figure this out.

**Comment by wlbragg** [comment]
Any Blender guru's skilled in channel manipulation watching this thread?

**Comment by wlbragg** [comment]
I've got the new key added to PanelTexturing-MainPanel.png.
![PanelTexturing-MainPanel](https://github.com/HHS81/c182s/assets/11069680/62cf1466-32f7-4432-9236-20fc43fb8f61)

I've added the new key added and mapped to interior-model-c182s.ac.
![Capture](https://github.com/HHS81/c182s/assets/11069680/5da90374-8faa-4d6d-88ee-5361d52728fe)

I have the new key lightmap RGB created.
![Capture](https://github.com/HHS81/c182s/assets/11069680/d7d43771-ab4e-44c9-96ac-09d6d82a8582)

All that is left is to add the alpha channel to the RGB channels and I can't figure it out.
RGB above + alpha below = new lightmap.
![Capture](https://github.com/HHS81/c182s/assets/11069680/2af002ed-b2c3-414a-8ec1-cba9105b6a7f)

Look at \c182s\Models\Effects\LMPanel.png, what I need to end up with is that 4 channel lightmap but with the new rgb edited channels. I'm following my own instructions in [Howto:Multi-channel_lightmap](https://wiki.flightgear.org/Howto:Multi-channel_lightmap) but it isn't working, I can't get the alpha channel applied correctly.

**Comment by TheFGFSEagle** [comment]
Hm, if you have an alpha layer and an RGB layer (without alpha channel, Menu -> CHannels -> Transparency -> remove alpha channel) the two should be mergeable without further issue ? How exactly is it not working ? Does the combined image have no alpha channel at all, is the alpha value set to the same value for the entire image, or do you get wrong alpha values ?

**Comment by wlbragg** [comment]
I think the issue is that it is not an alpha layer and I am not getting it converted to one. It either doesn't contain the needed data (the greyscale pixels at the bottom of the texture) or it isn't alpha where the data is not and blocks the rgb data when being used.

**Comment by wlbragg** [comment]
@TheFGFSEagle maybe you can figure this out.

This is the RGB channels
![LMPanel](https://github.com/HHS81/c182s/assets/11069680/70d449c0-8691-489d-ac21-927ea8764556)

This is what I need to add to the RGB channels as an alpha channel
![alpha](https://github.com/HHS81/c182s/assets/11069680/18598e04-f292-4e84-adb0-7fb96ac4bdfb)

**Comment by TheFGFSEagle** [comment]
Wait, does the original LMPanel also have an alpha channel ?

**Comment by TheFGFSEagle** [comment]
and you just added the key to both the alpha layer and the RGB layer ?

**Comment by wlbragg** [comment]
Not exactly. The original is a 4 layer lightmap.

R, G and B have the key and are used for...
dome-light-l
dome-light-r
glareshield-lights-norm

The alpha chanel is used for ...
pedestal-lights-norm
and  does not have the key.

**Comment by TheFGFSEagle** [comment]
Alright, I think (hope) I got it:
![rgba](https://github.com/HHS81/c182s/assets/77644149/fc9bf13d-813c-4804-a621-1e6e9b03e086)
After lots of searching and trying around, this worked out of the box: https://graphicdesign.stackexchange.com/a/158228/168007

**Comment by wlbragg** [comment]
Nice, thank you!

Are you kidding me. I saw the "decompose", but I thought that there was no way I was going to have to take those steps to do this. Guess I was wrong.

**Comment by TheFGFSEagle** [comment]
So is the texture what you needed ?

**Comment by wlbragg** [comment]
Yes, now we need to make or repair LMPanel-c182T.png. I haven't yet figured out exactly what is wrong with it.

**Comment by wlbragg** [comment]
@hbeni I decided to try to do the 182t lightmaps the proper way by setting it up in Blender and baking them. Do you know exactly where and how many glareshield lights there are?

**Comment by hbeni** [comment]
Sorry, I have no footage of the real thing. But I'm sure, Heiko researched this when making the model, so I think the best thing is to model it as a long single light like at the S variant.

**Comment by wlbragg** [comment]
I didn't even think to look and see if it is modeled. It is. I had no idea it was this big. It must be relatively diffuse.
![182s-panel-lights](https://github.com/HHS81/c182s/assets/11069680/84f27fbe-9031-4bbc-add1-df2c27c0be1e)
![182t-panel-lights](https://github.com/HHS81/c182s/assets/11069680/f4dc3f03-245a-4258-940d-074049f3f7a0)

**Comment by wlbragg** [comment]
>That is for the 182T, right?

Yeah, but it is in the 182S also. Same thing, no reference to it in any file except its existence in the .ac.

**Comment by hbeni** [comment]
partly merged [PR #532](https://github.com/HHS81/c182s/pull/532#issuecomment-1880486598)
> Merged fde0e5a and d62eda3; tested, looks OK.

The C182S is now OK. T variant still in the workshop

**Comment by wlbragg** [comment]
@hbeni 

OK, picking this back up. Do we have any images of how this lighting looks in real life? Specifically I am wondering where the glareshield lights stops illuminating and the pedestal light picks up? Also the tones we should expect? I'm searching now, but if you have any images or know of any, it would be helpful.

**Comment by wlbragg** [comment]
This is where I am currently at. I still have more objects to add and the pedestal lightmap needs to be fixed. I also wanted to look closer at the dome lightmaps. I still plan to adjust the intensity and the range of the light. What do you think about the color or tone of the light?

https://github.com/HHS81/c182s/assets/11069680/588e7b8f-b5c4-4d69-8d95-48c653154ecf

**Comment by hbeni** [comment]
I think the light tone is good :)

**Comment by hbeni** [comment]
Question: will this also fix #460 automatically?

**Comment by wlbragg** [comment]
Yes
